ir/constant.h

```diff
@@ -33,6 +34,16 @@ class IntConst final : public Constant {
   auto getInt() const { return std::get_if<int64_t>(&val); }
 };
 
+class FloatConst final : public Constant {
+  std::variant<float, double, std::string> val;
+public:
+  FloatConst(Type &type, float val);
+  FloatConst(Type &type, double val);
+  FloatConst(Type &type, std::string &&val);
```

Comment:
this path isn't supported in the SMT expr generation, is constructor called somewhere?  If not, pls delete it.